hack/ginkgo-e2e.sh

```diff
@@ -166,20 +175,41 @@ fi
 PATH=$(dirname "${e2e_test}"):"${PATH}"
 export PATH
 
-# Choose the program to execute.
-program=("${ginkgo}")
-if [[ "${E2E_TEST_DEBUG_TOOL}" == "delve" ]]; then
-  program=("dlv" "exec")
-elif [[ "${E2E_TEST_DEBUG_TOOL}" == "gdb" ]]; then
-  program=("gdb")
+# Choose the program to execute and additional arguments for it.
+#
+# Note that the e2e.test binary must have been built with "make DBG=1"
+# when using debuggers, otherwise debug information gets stripped.
+case "${E2E_TEST_DEBUG_TOOL:-ginkgo}" in
+  ginkgo)
+    program=("${ginkgo}" "${ginkgo_cli_args[@]}")
+    if [[ -n "${GINKGO_PARALLEL_NODES:-}" ]]; then
+      program+=("--nodes=${GINKGO_PARALLEL_NODES}")
+    elif [[ ${GINKGO_PARALLEL} =~ ^[yY]$ ]]; then
+      program+=("--nodes=25")
+    fi
```

Comment:
nit
```suggestion
    if [[ -n "${GINKGO_PARALLEL_NODES:-}" ]]; then
      ginkgo_args+=("--nodes=${GINKGO_PARALLEL_NODES}")
    elif [[ ${GINKGO_PARALLEL} =~ ^[yY]$ ]]; then
      ginkgo_args+=("--nodes=25")
    fi
```

Reply:
It doesn't make a difference here anymore, but I was trying to reserve `gingko_args` for those arguments that work in both places. I think adding to `program` is more aligned with that.